import "yd_fieldoptions.proto";
import "proto_common.proto";
import "proto_svr_common.proto";

package proto_ff;

// 服务器内部的消息id，是个统一的
// 
enum Proto_SvrMsgID
{
  NF_STSTORE_CHECK_STORE_SERVER_REQ = 40; //server to store server, check exist
  NF_STORETS_CHECK_STORE_SERVER_RSP = 41; //store server to server, check exist rsp
  NF_TEST_SEND_PROXY_MSG_TO_OTHER_SERVER_REQ = 42; //test send proxy msg to other server
  NF_TEST_OTHER_SERVER_SEND_MSG_TO_PROXY_SERVER = 43; //test send proxy msg to other server rsp
  NF_TEST_OTHER_SERVER_MSG_TO_WORLD_SERVER_REQ = 44; //test send other server to world server
  NF_TEST_WORLD_SERVER_MSG_TO_OTHER_SERVER_REQ = 45; //test send world msg to other server
}

message CommonMsgRsp
{
  optional int32 result = 1;
}

message Proto_STMasterServerDumpInfoNtf
{
  optional string dump_info = 1;
  optional uint32 bus_id = 2;
}

message Proto_MasterTMonitorReloadReq
{
  optional string server_name = 1;
  optional string server_id = 2;
}

message Proto_MasterTMonitorRestartReq
{
  optional string server_name = 1;
  optional string server_id = 2;
}

message Proto_MasterTMonitorStartReq
{
  optional string server_name = 1;
  optional string server_id = 2;
}

message Proto_MasterTMonitorStopReq
{
  optional string server_name = 1;
  optional string server_id = 2;
}

message Proto_MonitorTMasterReloadRsp
{
  optional int32 result = 1;
  optional string server_name = 2;
  optional string server_id = 3;
}

message Proto_MonitorTMasterRestartRsp
{
  optional int32 result = 1;
  optional string server_name = 2;
  optional string server_id = 3;
}


message Proto_MonitorTMasterStartRsp
{
  optional int32 result = 1;
  optional string server_name = 2;
  optional string server_id = 3;
}


message Proto_MonitorTMasterStopRsp
{
  optional int32 result = 1;
  optional string server_name = 2;
  optional string server_id = 3;
}

message Proto_STStoreCheckReq
{

}

message Proto_StoreTSCheckRsp
{

}

message Proto_TestSendProxyMsgToOtherServer
{
  optional string server_id = 1;
  optional string server_name = 2;
}

message Proto_TestOtherServerSendMsgToProxyServer
{
  optional string server_id = 3;
  optional string server_name = 4;
}

message Proto_TestSendWorldMsgToOtherServer
{
  optional string server_id = 1;
  optional string server_name = 2;
}

message Proto_TestOtherServerToWorldServer
{
  optional string server_id = 3;
  optional string server_name = 4;
}